### PR TITLE
Replace adminbar customize link with site-editor in FSE themes

### PR DESCRIPTION
**Pull request by aristath**
## Description
Replaces the "Customize" link with "Edit site".

This is only for FSE themes on the frontend.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by gziolo — COMMENTED** [review]
I saw that the other day, I'm sure there is more to tweak 👍

File: lib/full-site-editing.php [hunk 1]
```diff
@@ -57,6 +57,38 @@ function gutenberg_remove_legacy_pages() {
...
+	}
+
+	// Remove customizer link.
+	$wp_admin_bar->remove_node( 'customize' );
+	$wp_admin_bar->remove_node( 'customize-background' );
+	$wp_admin_bar->remove_node( 'customize-header' );
+	$wp_admin_bar->remove_node( 'widgets' );
+
+	// Add site-editor link.
+	if ( ! is_admin() ) {
```
Comment (gziolo) [inline lib/full-site-editing.php:79]: The logic for displaying link is way more complex in WP Core:

https://github.com/WordPress/wordpress-develop/blob/285fa03b3a57629df6a5dc9d92cc7751475c7dee/src/wp-includes/admin-bar.php#L419-L455

We might want to align. If we change it here, we probably should also update how it is exposed in the siebdar menu.

**Review by gziolo — APPROVED** [review]
The behavior proposed doesn't differ much from how the item in the sidebar is exposed:

https://github.com/WordPress/gutenberg/blob/1862c6274367c0c334f5a3362408537b5877121a/gutenberg.php#L105-L118

As a follow-up, we should limit this option based on user permissions.

Edit: Actually, there is an existing check for `edit_theme_options` capability, should we guard this link the same way?

File: lib/full-site-editing.php (hunk 1 above)
Comment (aristath) [reply lib/full-site-editing.php:79]: The WP-Core implementation is mostly for customizer changesets etc, so not something that can apply here. The only relevant part is the `is_admin` check and the permissions check which is missing here (commit coming in a few minutes)

**Comment by aristath** [comment]
> Edit: Actually, there is an existing check for `edit_theme_options` capability, should we guard this link the same way?

Good point. Added in https://github.com/WordPress/gutenberg/pull/27135/commits/e1cdbeec34afba685cf731dfd893b029b7f2cb99 :+1:

**Pull request merged**